src/Avalonia.Controls/Window.cs

```diff
@@ -125,7 +153,9 @@ static Window()
             BackgroundProperty.OverrideDefaultValue(typeof(Window), Brushes.White);
             TitleProperty.Changed.AddClassHandler<Window>((s, e) => s.PlatformImpl?.SetTitle((string)e.NewValue));
             HasSystemDecorationsProperty.Changed.AddClassHandler<Window>(
```

Comment:
It seems that `HasSystemDecorations` and `SystemDecorations` might get out of sync on the `Window`. @grokys any idea how to properly synchronize those?

Reply:
Do you mean `HasSystemDecorations` and the value on the platform impl? The pattern we're using elsewhere is exposing a property and a changed `Action` on the impl.

(I think we should use another mechanism but that's a different subject)

Reply:
There are now two avalonia properties that set the same platform property. Those will be out of sync.

Reply:
Ah, got you. Yeah, that's a bit of a difficult situation because both are styled properties. Perhaps we should change `HasSystemDecorations` to a direct property that sets `SystemDecorations`? It'd be a breaking change in the definition of `HasSystemDecorationsProperty` but not one that should cause a huge problem I hope?

We'd then define `HasSystemDecorations` as:

```
public bool HasSystemDecorations
{
    get => SystemDecorations != SystemDecorations.None;
    set => SystemDecorations = value ? SystemDecorations.Full : SystemDecorations.None;
}
```

Reply:
BTW, we talked a while ago about emitting warnings for obsolete properties from the XAML compiler. Would that be that difficult to do? Would help warn people about situations like this.

Reply:
There could be some existing code that uses data binding to `HasSystemDecorations` property.

Reply:
That should continue working. We'd have to raise `RaisePropertyChanged` on `HasSystemDecorations` when `SystemDecorations` changes but bindings should continue working as before.

Reply:
I've added a commit which fixes this but not sure if it's the preferred way?

@grokys suggestion is cleaner but I'm not sure how to implement it with avalonia's properties i.e. how to bind it so that setting SystemDecorations causes a property changed event for HasSystemDecorations and vice versa?